modules/standard/Types.chpl

```diff
@@ -711,4 +711,110 @@ inline proc _bor_id(type t) return 0:t;
 pragma "no doc"
 inline proc _bxor_id(type t) return 0:t;
 
+
+/* Return the number of fields in a class or record as a param.
+   The count of fields includes types and param fields.
+ */
+proc numFields(type t) param
```

Comment:
Did we establish some sort of rule of thumb as to when the ":returns:" field should be used and when it shouldn't, or did we leave it up to the author?  If the former, what was it?  If the latter, what's yours?